DateFieldMapper `ignore_malformed` misses java.lang.ArithmeticException

It seems when parsing dates as longs, we can have arithmetic exceptions. Steps to reproduce:

```
PUT test01
{
  "mappings": {
    "properties": {
      "dadate": {
        "type": "date",
        "ignore_malformed": true
      }
    }
  }
}
     
PUT test01/_doc/1
{
  "dadate": "-522000000"
}

```

The error in the logs looks like:

```
Caused by: java.lang.ArithmeticException: long overflow
	at java.lang.Math.multiplyExact(Math.java:946) ~[?:?]
	at java.lang.Math.multiplyExact(Math.java:922) ~[?:?]
	at java.time.Instant.toEpochMilli(Instant.java:1236) ~[?:?]
	at org.elasticsearch.index.mapper.DateFieldMapper$Resolution$1.convert(DateFieldMapper.java:83) ~[elasticsearch-7.6.0.jar:7.6.0]
	at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parse(DateFieldMapper.java:338) ~[elasticsearch-7.6.0.jar:7.6.0]
	at org.elasticsearch.index.mapper.DateFieldMapper.parseCreateField(DateFieldMapper.java:545) ~[elasticsearch-7.6.0.jar:7.6.0]

```


Similar to #50090

--- a/server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java
+++ b/server/src/main/java/org/elasticsearch/index/mapper/DateFieldMapper.java
@@ -518,7 +518,7 @@ public final class DateFieldMapper extends ParametrizedFieldMapper {
         } else {
             try {
                 timestamp = fieldType().parse(dateAsString);
-            } catch (IllegalArgumentException | ElasticsearchParseException | DateTimeException e) {
+            } catch (IllegalArgumentException | ElasticsearchParseException | DateTimeException | ArithmeticException e) {
                 if (ignoreMalformed) {
                     context.addIgnoredField(mappedFieldType.name());
                     return;

--- a/server/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java
+++ b/server/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java
@@ -167,9 +167,10 @@ public class DateFieldMapperTests extends ESSingleNodeTestCase {
                 "failed to parse date field [2016-03-99] with format [strict_date_optional_time||epoch_millis]");
         testIgnoreMalfomedForValue("-2147483648",
                 "Invalid value for Year (valid values -999999999 - 999999999): -2147483648");
+        testIgnoreMalfomedForValue("-522000000", "long overflow");
     }
 
-    private void testIgnoreMalfomedForValue(String value, String expectedException) throws IOException {
+    private void testIgnoreMalfomedForValue(String value, String expectedCause) throws IOException {
         String mapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type")
                 .startObject("properties").startObject("field").field("type", "date").endObject().endObject()
                 .endObject().endObject());
@@ -185,7 +186,9 @@ public class DateFieldMapperTests extends ESSingleNodeTestCase {
                         .endObject()),
                 XContentType.JSON));
         MapperParsingException e = expectThrows(MapperParsingException.class, runnable);
-        assertThat(e.getCause().getMessage(), containsString(expectedException));
+        assertThat(e.getMessage(), containsString("failed to parse field [field] of type [date]"));
+        assertThat(e.getMessage(), containsString("Preview of field's value: '" + value + "'"));
+        assertThat(e.getCause().getMessage(), containsString(expectedCause));
 
         mapping = Strings.toString(XContentFactory.jsonBuilder().startObject().startObject("type")
                 .startObject("properties").startObject("field").field("type", "date")

